implementations/rust/ockam/ockam_command/src/node/show.rs

```diff
@@ -180,77 +63,51 @@ pub async fn print_query_status(
-        let services: ServiceList = node.ask(ctx, api::list_services()).await?;
+    let node_info =
+        if !is_node_up(ctx, node_name, node, cli_state.clone(), wait_until_ready).await? {
+            // it is expected to not be able to open an arbitrary TCP connection on an authority node
+            // so in that case we display an UP status
+            let is_authority_node = node_state.config().setup().authority_node.unwrap_or(false);
+            ShowNodeResponse::new(is_default, node_name, is_authority_node, node_port)
+        } else {
+            let mut node_info = ShowNodeResponse::new(is_default, node_name, true, node_port);
 
-        // Get list of TCP listeners for node
-        let tcp_listeners: TransportList = node.ask(ctx, api::list_tcp_listeners()).await?;
+            // Get short id for the node
+            node_info.identity = Some(match node_state.config().identity_config() {
+                Ok(resp) => resp.identifier().to_string(),
+                Err(_) => String::from("None"),
+            });
 
-        // Get list of Secure Channel Listeners
-        let secure_channel_listeners: SecureChannelListenersList =
-            node.ask(ctx, api::list_secure_channel_listener()).await?;
+            // Get list of services for the node
+            node_info.service_list = Some(node.ask(ctx, api::list_services()).await?);
 
-        // Get list of inlets
-        let inlets: InletList = node.ask(ctx, api::list_inlets()).await?;
+            // Get list of TCP listeners for node
+            node_info.tcp_listeners = Some(node.ask(ctx, api::list_tcp_listeners()).await?);
 
-        // Get list of outlets
-        let outlets: OutletList = node.ask(ctx, api::list_outlets()).await?;
+            // Get list of Secure Channel Listeners
+            node_info.secure_channel_listeners =
+                Some(node.ask(ctx, api::list_secure_channel_listener()).await?);
 
-        let node_state = cli_state.nodes.get(node_name)?;
-        let node_port = node_state
-            .config()
-            .setup()
-            .api_transport()
-            .ok()
-            .map(|listener| listener.addr.port());
-
-        print_node_info(
-            opts,
-            node_port,
-            node_name,
-            is_default,
-            true,
-            Some(&default_id),
-            Some(&services),
-            Some(&tcp_listeners),
-            Some(&secure_channel_listeners),
-            Some((&inlets, &outlets)),
-        );
-    }
+            // Get list of inlets
+            let inlets: InletList = node.ask(ctx, api::list_inlets()).await?;
+            node_info.inlets = Some(inlets);
 
-    Ok(())
+            // Get list of outlets
+            let outlets: OutletList = node.ask(ctx, api::list_outlets()).await?;
+            node_info.outlets = Some(outlets);
+
+            node_info
+        };
+
+    Ok(opts.global_args.output_format.println_value(&node_info)?)
```

Comment:
Since you now have an instance that implements the `Output` trait, the correct way to print the command output is throught the `terminal` instance. It would look something like:

```rust
opts.terminal
    .stdout()
    .plain(node_info)
    .json(serde_json::to_string_pretty(&node_info)?)
    .write_line()?;
```


[Here is an exampe](https://github.com/build-trust/ockam/blob/1a49eba84f17ea3b368aa22a1ffc6e53a474b474/implementations/rust/ockam/ockam_command/src/relay/delete.rs#L58) of another command using it.

Reply:
Fixed